graphql_compiler/compiler/metadata.py

```diff
@@ -132,6 +154,27 @@ def get_location_info(self, location):
                                  u'{}'.format(location))
         return location_info
 
+    def record_filter_info(self, location, filter_info):
+        """Record filter information about the location."""
+        if isinstance(location, FoldScopeLocation):
+            # NOTE(gurer): ignore filters inside the fold for now
+            return
+        record_location = location.at_vertex()
+        self._filter_infos.setdefault(record_location, []).append(filter_info)
+
+    def get_filter_infos(self, location):
+        """Get information about filters at the location."""
+        return self._filter_infos.get(location, [])
+
+    def record_recurse_info(self, location, recurse_info):
+        """Record recursion information about the location."""
+        record_location = location.at_vertex()
```

Comment:
I think this behavior is correct, because different visit counters signify different stages of the evaluation of the data at a given location, and I'd prefer to capture that info accurately in metadata.